### PR TITLE
Add Presentation Layer for UserProfileService #5

### PR DESCRIPTION
**Pull request by vladzhorov**

### CODE REVIEW

File: ServiceFinder/ServiceFinder.API/Controller/AssistanceCategoryController.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            var assistance = _mapper.Map<AssistanceCategory>(viewModel);
+            var result = await _AssistanceCategoryService.CreateAsync(assistance, cancellationToken);
+            return _mapper.Map<AssistanceCategoryViewModel>(result);
+        }
+
+        [HttpPut("{id}")]
+        public async Task<AssistanceCategoryViewModel> Update(Guid id, UpdateAssistanceCategoryViewModel viewModel, CancellationToken cancellationToken)
+        {
+            var modelToUpdate = _mapper.Map<AssistanceCategory>(viewModel);
+            modelToUpdate.Id = id;
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/AssistanceCategoryController.cs:48]: is this Presentation layer responsibility?

File: ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route("api/assistances")]
+    public class AssistanceController : ControllerBase
+    {
+        private readonly IAssistanceService _AssistanceService;
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs:13]: naming: _A

File: ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            var assistance = _mapper.Map<Assistance>(viewModel);
+            var result = await _AssistanceService.CreateAsync(assistance, cancellationToken);
+            return _mapper.Map<AssistanceViewModel>(result);
+        }
+
+        [HttpPut("{id}")]
+        public async Task<AssistanceViewModel> Update(Guid id, UpdateAssistanceViewModel viewModel, CancellationToken cancellationToken)
+        {
+            var modelToUpdate = _mapper.Map<Assistance>(viewModel);
+            modelToUpdate.Id = id;
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs:48]: same

File: ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        public async Task<List<ReviewViewModel>> GetAll(CancellationToken cancellationToken)
+        {
+            var Reviews = await _ReviewService.GetAllAsync(cancellationToken);
+            return _mapper.Map<List<ReviewViewModel>>(Reviews);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ReviewViewModel> GetById(Guid id, CancellationToken cancellationToken)
+        {
+            var Review = await _ReviewService.GetByIdAsync(id, cancellationToken);
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs:32]: naming

File: ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route("api/reviews")]
+    public class ReviewController : ControllerBase
+    {
+        private readonly IReviewService _ReviewService;
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs:13]: naming

File: ServiceFinder/ServiceFinder.API/Controller/UserProfileController.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using ServiceFinder.API.ViewModels.UserProfile;
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route("api/usersProfile")]
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/UserProfileController.cs:10]: move all routes to constants

File: ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace ServiceFinder.API.Mapper
+{
+    internal class Mapping : Profile
+    {
+        public Mapping()
+        {
+            CreateMap<Assistance, AssistanceViewModel>().ReverseMap();
+            CreateMap<CreateAssistanceViewModel, Assistance>()
+                .ForMember(dest => dest.CreatedAt, opt => opt.MapFrom(src => DateTime.UtcNow))
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs:16]: ??????????????????

File: ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+                .ForMember(dest => dest.UpdatedAt, opt => opt.MapFrom(src => DateTime.UtcNow));
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs:17]: ??????????????????????

File: ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+            CreateMap<UpdateAssistanceViewModel, Assistance>()
+                .ForMember(dest => dest.UpdatedAt, opt => opt.MapFrom(src => DateTime.UtcNow));
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs:19]: ???????????????????

File: ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+            CreateMap<CreateReviewViewModel, Review>()
+                .ForMember(dest => dest.CreatedAt, opt => opt.MapFrom(src => DateTime.UtcNow))
+                .ForMember(dest => dest.UpdatedAt, opt => opt.MapFrom(src => DateTime.UtcNow));
+
+            CreateMap<UserProfile, UserProfileViewModel>().ReverseMap();
+            CreateMap<CreateUserProfileViewModel, UserProfile>()
+                .ForMember(dest => dest.CreatedAt, opt => opt.MapFrom(src => DateTime.UtcNow))
+                .ForMember(dest => dest.UpdatedAt, opt => opt.MapFrom(src => DateTime.UtcNow));
+            CreateMap<UpdateUserProfileViewModel, UserProfile>()
+                .ForMember(dest => dest.UpdatedAt, opt => opt.MapFrom(src => DateTime.UtcNow));
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Mapper/Mapping.cs:38]: wtf

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: ServiceFinder/ServiceFinder.API/Controller/AssistanceCategoryController.cs [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route(ApiRoutes.assistancesCategory)]
+    public class AssistanceCategoryController : ControllerBase
+    {
+        private readonly IAssistanceCategoryService _AssistanceCategoryService;
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/AssistanceCategoryController.cs:14]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.API/Controller/AssistanceCategoryController.cs [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.AspNetCore.Mvc;
+using ServiceFinder.API.DI;
+using ServiceFinder.API.ViewModels.AssistanceCategory;
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route(ApiRoutes.assistancesCategory)]
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/AssistanceCategoryController.cs:11]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.AspNetCore.Mvc;
+using ServiceFinder.API.DI;
+using ServiceFinder.API.ViewModels.Assistance;
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route(ApiRoutes.assistances)]
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs:11]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+        public async Task<List<AssistanceViewModel>> GetAll(CancellationToken cancellationToken)
+        {
+            var Assistances = await _assistanceService.GetAllAsync(cancellationToken);
+            return _mapper.Map<List<AssistanceViewModel>>(Assistances);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<AssistanceViewModel> GetById(Guid id, CancellationToken cancellationToken)
+        {
+            var Assistance = await _assistanceService.GetByIdAsync(id, cancellationToken);
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/AssistanceController.cs:33]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+        public async Task<List<ReviewViewModel>> GetAll(CancellationToken cancellationToken)
+        {
+            var Reviews = await _reviewService.GetAllAsync(cancellationToken);
+            return _mapper.Map<List<ReviewViewModel>>(Reviews);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ReviewViewModel> GetById(Guid id, CancellationToken cancellationToken)
+        {
+            var Review = await _reviewService.GetByIdAsync(id, cancellationToken);
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs:33]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+using Microsoft.AspNetCore.Mvc;
+using ServiceFinder.API.DI;
+using ServiceFinder.API.ViewModels.Review;
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route(ApiRoutes.reviews)]
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/ReviewController.cs:11]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.API/Controller/UserProfileController.cs [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+using Microsoft.AspNetCore.Mvc;
+using ServiceFinder.API.DI;
+using ServiceFinder.API.ViewModels.UserProfile;
+using ServiceFinder.BLL.Abstarctions.Services;
+using ServiceFinder.BLL.Models;
+
+namespace ServiceFinder.API.Controller
+{
+    [ApiController]
+    [Route(ApiRoutes.usersProfile)]
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/Controller/UserProfileController.cs:11]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.API/DI/ApiRoutes.cs [hunk 18]
```diff
@@ -0,0 +1,10 @@
+﻿namespace ServiceFinder.API.DI
+{
+    public class ApiRoutes
+    {
+        public const string assistancesCategory = "api/assistancesCategory";
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/DI/ApiRoutes.cs:5]: "api" can be moved to separate variable

File: ServiceFinder/ServiceFinder.API/DI/ApiRoutes.cs [hunk 19]
```diff
@@ -0,0 +1,10 @@
...
+        public const string assistances = "api/assistances";
+        public const string reviews = "api/reviews";
+        public const string usersProfile = "api/usersProfile";
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.API/DI/ApiRoutes.cs:8]: naming conventions are not followed

File: ServiceFinder/ServiceFinder.DAL/Entites/BaseEntity.cs [hunk 20]
```diff
@@ -0,0 +1,9 @@
+﻿using ServiceFinder.DAL.Interfaces;
+
+namespace ServiceFinder.DAL.Entites
+{
+    public class BaseEntity : IBaseEntity
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Entites/BaseEntity.cs:5]: what's the reason to add both class and interface?

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

**Review by VadimZharikov — APPROVED** [review]

**Pull request merged**